Frameworks/UIKit/UIImage.mm

```diff
@@ -215,348 +209,213 @@ + (id)imageWithCGImage:(CGImageRef)image scale:(float)scaleFactor orientation:(U
-    UINT width = 0, height = 0;
-    HRESULT hr = S_OK;
-
-    hr = pFrame->GetSize(&width, &height);
-
-    if (SUCCEEDED(hr)) {
-        CGColorSpaceRef clrRgb = CGColorSpaceCreateDeviceRGB();
-        dest->m_pImage =
-            CGImageCreate(width, height, 8, 32, width * 4, clrRgb, kCGImageAlphaLast, nil, NULL, false, kCGRenderingIntentDefault);
-        CGColorSpaceRelease(clrRgb);
-
-        hr = pFactory->CreateFormatConverter(&pFormatConverter);
-        if (SUCCEEDED(hr)) {
-            hr = pFormatConverter
-                     ->Initialize(pFrame, GUID_WICPixelFormat32bppRGB, WICBitmapDitherTypeNone, NULL, 0.f, WICBitmapPaletteTypeCustom);
-            if (SUCCEEDED(hr)) {
-                BYTE* imageData = (BYTE*)dest->m_pImage->Backing()->LockImageData();
-                hr = pFormatConverter->CopyPixels(NULL,
-                                                  dest->m_pImage->Backing()->BytesPerRow(),
-                                                  dest->m_pImage->Backing()->BytesPerRow() * dest->m_pImage->Backing()->Height(),
-                                                  imageData);
-                dest->m_pImage->Backing()->ReleaseImageData();
-
-                if (SUCCEEDED(hr)) {
-                    ret = true;
-                } else {
-                    TraceError(TAG, L"IWICFormatConverter::CopyPixels failed hr=%x", hr);
-                    ret = false;
-                }
-            } else {
-                TraceError(TAG, L"IWICFormatConverter::Initialize failed hr=%x", hr);
-                ret = false;
-            }
-        } else {
-            TraceError(TAG, L"IWICImagingFactory::CreateFormatConverter failed hr=%x", hr);
-            ret = false;
-        }
-    } else {
-        TraceError(TAG, L"IWICBitmapDecoder::GetFrame failed hr=%x", hr);
-        ret = false;
-    }
-
-    if (pFormatConverter) {
-        pFormatConverter->Release();
+- (instancetype)init {
+    if (self = [super init]) {
+        _scale = 1.0f;
+        _imageStretch.origin.x = 0.0f;
+        _imageStretch.origin.y = 0.0f;
+        _imageStretch.size.width = 1.0f;
+        _imageStretch.size.height = 1.0f;
     }
-
-    return ret;
+    return self;
 }
 
-static bool loadImageWithWICDecoder(UIImage* dest, REFGUID decoderCls, void* bytes, int length) {
-    IWICImagingFactory* pFactory = NULL;
-    IWICBitmapDecoder* pDecoder = NULL;
-    IStream* spStream = NULL;
-    IWICBitmapFrameDecode* pFrame = NULL;
-    HRESULT hr = S_OK;
-
-    MULTI_QI mq = { 0 };
-
-    mq.pIID = &IID_IWICImagingFactory;
-    hr = CoCreateInstanceFromApp(CLSID_WICImagingFactory, NULL, CLSCTX_INPROC_SERVER, NULL, 1, &mq);
-
-    if (SUCCEEDED(hr)) {
-        pFactory = (IWICImagingFactory*)mq.pItf;
-        hr = pFactory->CreateDecoder(decoderCls, NULL, &pDecoder);
+/**
+ @Status Interoperable
+*/
+- (instancetype)initWithCGImage:(CGImageRef)image {
+    if (!image) {
+        [self release];
+        return nil;
     }
```

Comment:
Yes

---
In reply to: [84749461](https://github.com/Microsoft/WinObjC/pull/1185#discussion_r84749461) [](ancestors = 84749461)